apps/ejabberd/src/mod_mam_utils.erl

```diff
@@ -346,6 +346,25 @@ is_valid_message_children(_,     false, false, false) -> true;
 %% See mam_SUITE:offline_message for a test case
 is_valid_message_children(_,      _,    _,     _    ) -> false.
 
+is_chat_marker(Packet) ->
+    case mod_mam_params:archive_chat_markers() of
+        true ->
+            do_check_markers(Packet);
+        false ->
+            false
+    end.
+
+do_check_markers(Packet) ->
+    MarkerTags = [<<"received">>, <<"displayed">>, <<"acknowledged">>],
+    HasSubtag = fun(Tag) ->
+                    case xml:get_subtag(Packet, Tag) of
```

Comment:
Nope. It's feature freeze Monday so I'm going to create beta2 branch a bit later, to avoid "maybe one PR more... maybe another one... and maybe this one...." syndrome. :)